### PR TITLE
Do not throw for user paths that contain a quote character in them.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://devdiv.visualstudio.com/DevDiv/_queries/edit/1436188

### CODE REVIEW

File: src/VisualStudio/CSharp/Test/PersistentStorage/SQLiteV2PersistentStorageTests.cs [hunk 1]
```diff
@@ -46,7 +46,7 @@ public async Task TestCrashInNewConnection()
 
             // Because instantiating the connection will fail, we will not get back
             // a working persistent storage. We are testing a fault recovery code path.
-            await using (var storage = await GetStorageAsync(solution, faultInjector, throwOnFailure: false))
+            await using (var storage = await GetStorageAsync(solution, faultInjector: faultInjector, throwOnFailure: false))
```
Comment (ryzngard) [inline src/VisualStudio/CSharp/Test/PersistentStorage/SQLiteV2PersistentStorageTests.cs:49]: Why?

**Review by ryzngard — APPROVED** [review]

File: src/VisualStudio/CSharp/Test/PersistentStorage/SQLiteV2PersistentStorageTests.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/CSharp/Test/PersistentStorage/SQLiteV2PersistentStorageTests.cs:49]: i added another optional parameter, and this positional one needed to become named.

**Comment by dibarbet** [comment]
@CyrusNajmabadi does this need to be taken for servicing / backported?  Or is 17.1p3 good enough

**Comment by CyrusNajmabadi** [comment]
We've lived with it thsi far.  I imagine we don't need servicing.  (it's just degraded perf, not any other user visible problem)

**Pull request merged**